logger.go

```diff
@@ -27,65 +27,84 @@ import (
 	"log"
 	"net/http"
 	"os"
+	"path/filepath"
+	"strings"
 )
 
 type LoggingClient struct {
-	owningServiceName string
-	RemoteUrl         string
-	LogFilePath       string
-	stdOutLogger      *log.Logger
-	fileLogger        *log.Logger
+	OwningServiceName string
+	RemoteEnabled     bool
+	LogTarget         string
+	StdOutLogger      *log.Logger
+	FileLogger        *log.Logger
 }
 
 // Create a new logging client for the owning service
-func NewClient(owningServiceName string, remoteUrl string) LoggingClient {
+func NewClient(owningServiceName string, isRemote bool, logTarget string) LoggingClient {
 	// Set up logging client
 	lc := LoggingClient{
-		owningServiceName: owningServiceName,
-		RemoteUrl:         remoteUrl,
+		OwningServiceName: owningServiceName,
+		RemoteEnabled:     isRemote,
+		LogTarget:         logTarget,
 	}
 
 	// Set up the loggers
-	lc.stdOutLogger = log.New(os.Stdout, "", log.Ldate|log.Ltime|log.Lshortfile)
-	lc.fileLogger = &log.Logger{}
-	lc.fileLogger.SetFlags(log.Ldate | log.Ltime | log.Lshortfile)
-
-	// Default path
-	lc.LogFilePath = ""
+	lc.StdOutLogger = log.New(os.Stdout, "", log.Ldate|log.Ltime|log.Lshortfile)
+	lc.FileLogger = &log.Logger{}
+	lc.FileLogger.SetFlags(log.Ldate | log.Ltime | log.Lshortfile)
 
 	return lc
 }
 
 // Send the log out as a REST request
 func (lc LoggingClient) log(logLevel support_domain.LogLevel, msg string, labels []string) error {
-	// TODO: Find out if the log type is loggable (is it enabled)
 
-	// Save to logging file if path was set
-	lc.saveToLogFile(string(logLevel), msg)
+    if !lc.RemoteEnabled {
+		// Save to logging file if path was set
+		return lc.saveToLogFile(string(logLevel), msg)
+	}
 
 	// Send to logging service
 	logEntry := lc.buildLogEntry(logLevel, msg, labels)
 	return lc.sendLog(logEntry)
 }
 
-func (lc LoggingClient) saveToLogFile(prefix string, message string) {
-	if lc.LogFilePath != "" {
-		file, err := os.OpenFile(lc.LogFilePath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
-		defer file.Close()
-		if err != nil {
-			fmt.Println("Error opening log file: " + err.Error())
-		}
+func (lc LoggingClient) saveToLogFile(prefix string, message string) error {
+	if lc.LogTarget == "" {
+		return nil
+	}
+
+	verifyLogDirectory(lc.LogTarget)
```

Comment:
Can be the call to verifyLogDirectory to NewClient?

Reply:
You're saying put this in the constructor? OK, let me look at doing that.